perfkitbenchmarker/rackspace/rackspace_disk.py

```diff
@@ -0,0 +1,209 @@
+    """Boot disk is always attached."""
+    pass
+
+  def Detach(self):
+    """No need to detach boot disk."""
+    pass
+
+  def GetDevicePath(self):
+    return '/dev/%s' % self.device_name
+
+
+class RackspaceRemoteDisk(disk.BaseDisk):
+  """Object representing a Rackspace Volume."""
+
+  def __init__(self, disk_spec, name, zone, image=None):
+      super(RackspaceRemoteDisk, self).__init__(disk_spec)
+      self.attached_vm_name = None
+      self.image = image
+      self.name = name
+      self.zone = zone
+      self.id = ''
+      self.device_letter = None
+      self.num_retried_attach = 0
+
+  def __str__(self):
+    return ("%s(device_path=%s, id=%s, attached_vm=%s, disk_type=%s)"
+            % (self.__class__.__name__, self.GetDevicePath(),
+               self.id, self.attached_vm_name, self.disk_type))
+
+  def __repr__(self):
+    return self.__str__()
+
+  def _Create(self):
+    """Creates the volume."""
+    nova_env = os.environ.copy()
+    nova_env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+    create_cmd = [
+        FLAGS.nova_path,
+        'volume-create',
+        '--display-name', self.name,
+        '--volume-type', DISK_TYPE[self.disk_type],
+        str(self.disk_size)]
+    stdout, _, _ = vm_util.IssueCommand(create_cmd, env=nova_env)
+    volume = util.ParseNovaTable(stdout)
+    if 'id' in volume and volume['id']:
+      self.id = volume['id']
+    else:
+      raise errors.Error('There was a problem when creating a volume.')
+
+  def _Delete(self):
+    """Deletes the volume."""
+    nova_env = os.environ.copy()
+    nova_env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+    delete_cmd = [FLAGS.nova_path, 'volume-delete', self.name]
+    vm_util.IssueCommand(delete_cmd, env=nova_env)
+
+  def _Exists(self):
+    """Returns true if the volume exists."""
+    nova_env = os.environ.copy()
+    nova_env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+    getdisk_cmd = [FLAGS.nova_path, 'volume-show', self.name]
+    stdout, _, _ = vm_util.IssueCommand(getdisk_cmd, env=nova_env)
+    if stdout.strip() == '':
+      return False
+    volume = util.ParseNovaTable(stdout)
+    return 'status' in volume and volume['status'] == 'available'
+
+  def Attach(self, vm):
+    """Attaches the volume to a VM.
+
+    Args:
+      vm: The RackspaceVirtualMachine instance to which the volume will be
+          attached.
+    """
+    self.attached_vm_name = vm.name
+    nova_env = os.environ.copy()
+    nova_env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+    attach_cmd = [
+        FLAGS.nova_path,
+        'volume-attach',
+        self.attached_vm_name,
+        self.id, ""]
+    vm_util.IssueRetryableCommand(attach_cmd, env=nova_env)
+
+    volume_show_cmd = [FLAGS.nova_path, 'volume-show', self.name]
+
+    attached = False
+    for i in xrange(12):
+      stdout, _, _ = vm_util.IssueCommand(volume_show_cmd, env=nova_env)
+      attrs = stdout.split('\n')
+      for attr in attrs[3:-2]:
```

Comment:
This is rather hard to follow without example output. Are the initial and trailing line counts that it skips guaranteed to work this way? Depending on how the split below works, would it be ok to just loop over everything?